akka-stream/src/main/scala/akka/stream/impl/StreamLayout.scala

```diff
@@ -311,7 +311,16 @@ private[stream] class VirtualSubscriber[T](val owner: VirtualPublisher[T]) exten
  */
 private[stream] class VirtualPublisher[T]() extends Publisher[T] {
   @volatile var realPublisher: Publisher[T] = null
-  override def subscribe(s: Subscriber[_ >: T]): Unit = realPublisher.subscribe(s)
+  override def subscribe(s: Subscriber[_ >: T]): Unit = {
+    val sub = realPublisher.subscribe(s)
+    realPublisher match {
+      case _: SupportingMultipleSubscribers   ⇒
```

Comment:
I still don’t understand this point: the `fanoutPublisher` is a Sink that materializes to a Publisher that supports multiple subscribers. What we are discussing here has nothing to do with this, it is about how we pass along Publishers internally. The output is again determined by the Sink the user attaches, and if that is a `Sink.publisher`, why should it ever support multiple subscriptions?